src/app/data-planes/components/DataplanePolicies.vue

```diff
@@ -250,80 +252,103 @@ function getPolicyEntryConnections(policy: MatchedPolicyType, policyType: Policy
 
-    policyEntry.connections.push(...connections)
+    policyTypeEntry.connections.push(...connections)
   }
 
-  const policyTypeEntries = Array.from(policyEntriesByType.values())
+  const policyTypeEntries = Array.from(policyTypeEntriesByType.values())
 
-  policyTypeEntries.sort((policyEntryA, policyEntryB) => policyEntryA.type.localeCompare(policyEntryB.type))
+  policyTypeEntries.sort((policyTypeEntryA, policyTypeEntryB) => policyTypeEntryA.type.localeCompare(policyTypeEntryB.type))
 
   return policyTypeEntries
 }
 
-function getPolicyEntryConnectionsFromRules(rule: DataplaneRule, policyType: PolicyType): PolicyTypeEntryConnection[] {
+function getRuleEntryConnections(rule: DataplaneRule, policyType: PolicyType): RuleEntryConnection[] {
   const { type, service, subset, conf } = rule
 
   // Guards against likely API changes. The response currently contains `"subset": {}` instead of omitting the value or setting it to `null`, but that might change in the future.
   const subsetEntries = subset ? Object.entries(subset) : []
   let sourceTags: LabelValue[] | null
   let destinationTags: LabelValue[] | null
 
+  // Determines source tags.
   if (type === 'clientSubset') {
+    // For client subsets, the source is represented by `subset` (i.e. tags)
     // Sets the wildcard service tag for client subsets without subset values.
     const sourceSubset = subsetEntries.length > 0 ? subsetEntries : [['kuma.io/service', '*']]
-    // For client subsets, the source is represented by `subset` (i.e. tags); for destination subsets and single items, the source is the DPP.
     sourceTags = sourceSubset.length > 0 ? sourceSubset.map(([label, value]) => ({ label, value })) : null
   } else {
+    // For destination subsets and single items, the source is the DPP for which we don’t want to show anything.
     sourceTags = null
   }
 
-  if (typeof service === 'string' && service !== '') {
+  // Determines destination tags.
+  if (type === 'destinationSubset') {
+    // For destination subsets, the destination is represented by either `subset` (which has priority) or `service`. `subset` is potentially more specific than `service`.
+    if (subsetEntries.length > 0) {
+      destinationTags = subsetEntries.map(([label, value]) => ({ label, value }))
+    } else if (typeof service === 'string' && service !== '') {
+      // The `service` field, when set, represents the name of the destination service of traffic.
+      destinationTags = [
+        {
+          label: 'kuma.io/service',
+          value: service,
+        },
+      ]
+    } else {
+      // For destination subsets with empty or absent `subset` field and empty or absent `service`, we can set the wildcard service tag to indicate that the traffic goes to all services.
+      destinationTags = [
+        {
+          label: 'kuma.io/service',
+          value: '*',
+        },
+      ]
+    }
+  } else if (type === 'clientSubset' && typeof service === 'string' && service !== '') {
```

Comment:
Is there any reason not to include this in the `type === 'clientSubset'` fork higher up (starting at line 312)?

Reply:
Yes, I want to separate setting source tags from setting destination tags because that makes it easier to follow the individual conditions.